packages/node/src/behaviors/thermostat/ThermostatServer.ts

```diff
@@ -4,14 +4,1000 @@
+    Thermostat.Feature.MatterScheduleConfiguration,
+);
+
 /**
  * This is the default server implementation of {@link ThermostatBehavior}.
  *
  * The Matter specification requires the Thermostat cluster to support features we do not enable by default. You should
  * use {@link ThermostatServer.with} to specialize the class for the features your implementation supports.
+ *
+ * This implementation mainly provides all validation and base logic required by the Matter specification.
+ * It **does not** implement any logic to actually control a thermostat device or to react to temperature changes to e.g.
+ * adjust the ThermostatRunningMode when the Auto feature is supported. This is all up to the specific implementation.
+ *
+ * For local temperature or occupancy values we check if there is a local cluster available on the same endpoint and use
+ * them, alternatively raw measurements can be set in the states externalMeasuredIndoorTemperature and
+ * externallyMeasuredOccupancy. The OutdoorTemperature can be set directly on the attribute if supported.
+ * The RemoteSensing attribute need to be set correctly as needed by the developer to identify the measurement source.
  */
-export class ThermostatServer extends ThermostatBehavior {}
+export class ThermostatBaseServer extends ThermostatBehaviorLogicBase {
+    declare protected internal: ThermostatBaseServer.Internal;
+    declare state: ThermostatBaseServer.State;
+    declare events: ThermostatBaseServer.Events;
+
+    override initialize() {
+        if (this.features.scheduleConfiguration) {
+            throw new ImplementationError("ScheduleConfiguration features is deprecated and not allowed to be used!");
+        }
+        if (this.features.presets || this.features.matterScheduleConfiguration) {
+            logger.warn(
+                "Presets and MatterScheduleConfiguration features are not yet implemented. Please do not activate them.",
+            );
+        }
+
+        this.#setupValidations();
+        this.#setupLocalTemperatureMeasurementIntegration();
+        this.#setupOccupancyIntegration();
+        this.#setupModeHandling();
+        this.#setupThermostatLogic();
+        this.#setupSetback();
+
+        // Fix invalid default we have internally and ensure correct value for now, we update model with 1.4.2 update
+        if (this.state.minSetpointDeadBand === 2000) {
+            this.state.minSetpointDeadBand = 200;
+        }
+        if (this.state.minSetpointDeadBand < 0 || this.state.minSetpointDeadBand > 127) {
+            throw new ImplementationError("minSetpointDeadBand is out of valid range 0..127");
+        }
+        // Store these states internally to prevent changes
+        this.internal.minSetpointDeadBand = this.state.minSetpointDeadBand;
+        this.internal.controlSequenceOfOperation = this.state.controlSequenceOfOperation;
+    }
+
+    override setpointRaiseLower({ mode, amount }: Thermostat.SetpointRaiseLowerRequest) {
```

Comment:
Consider breaking up into functions, one for each mode, so this method is just a single switch + function calls